### DISCUSSION TITLE
Custom Tabufline tab format and width [full path]

### DISCUSSION THREAD

**Discussion by xander412** [NvChad/NvChad#2944] [Q&A] [answered]
I think this is very basic and silly question, but I am just getting started. Need custom tabufline tab name format, I need to see the filename always and the path should only wrap only when the tabline crosses the window width.  And also please share any method through which I can see all the options that any setup or function is having. Is code browsing only the option, or I have some way to directly get all the options like dir method in python or similar. So, that I don't post too many questions online :)

**Comment by siduck** [comment]
read `:h stl` and there are some special letters   (along with %letter)  which will be parsed by statusline or tabline to show output

ofr example `  %F ` will show full filename along with spaces around it.

please read how to override tabufline modules.. https://nvchad.com/docs/config/nvchad_ui/

**Reply by xander412** [reply]
Hi @siduck . Thanks for the quick response. I tried the options, but when I set tabline=<some format>, its only showing the current filename overriding everything else. I tried going through the tabufline override methods, but M.ui = {
  tabufline = {
    --  more opts
    order = { "treeOffset", "buffers", "tabs", "btns", 'abc' },
    modules = {
      -- You can add your custom component
      abc = function()
        return "hi"
      end,
    }
  },
. Here, can I override the buffers method? If yes could u tell show an example buffer method override, because I want loop through every opened tab and assign the strings appropriately. Or please point me towards the source code of "buffers" which I can change statically.

**Reply by siduck** [reply]
You should make new module that does that and mention the module name in the order...

**Reply by xander412** [reply]
I am completely new to lua. I can make  a dummy module, but what is need the module to do is : 

new_module = function()
string = ""
for (buf_idx = 1 --> buf_idx = end)
    get_curr_buf_idx = vim.api.nvim_set_current_buf(vim.t.bufs[buf_idx])
    actual_string += filename(get_curr_buf_idx, "format")
end
return actual_string
end

Now, order = {"treeOffset", "new_module", ...}.

Obvisouly, I don't know the exact methods to call. And also how to set the required format. Sorry, but I am completely. I just need the tabs to be in a specific format that is specified in :h stl. Thank you

**Reply by siduck** [reply]
sadly you will need to define whole buffers module :/

**Reply by xander412** [reply]
Could you point to the existing source code such that I can write the buffers module.

 My main problem is that I need the full name of the file to navigate between the files and I am very used to this. Or is there any variable that controls the width of the tab.

**Reply by siduck** [reply]
the fullname will just make the bufferline filled by just 1/2 files :/ 

it will be easier to show fullname on the statusline instead

**Reply by xander412** [reply]
Full name in the sense not the absolute path, just the name of the file which is about 20-25 characters

**Reply by siduck** [reply, marked as answer]
hmm that'll still be hard, cuz you'll have to write the  whole buffers module.  check the tabufline modules file

https://github.com/NvChad/ui/blob/v2.5/lua/nvchad/tabufline/modules.lua

**Reply by xander412** [reply]
Thank you very much
